certbot/client.py

```diff
@@ -410,11 +410,13 @@ def enhance_config(self, domains, config, chain_path):
             logger.warning("No config is specified.")
             raise errors.Error("No config available")
 
-        supported = self.installer.supported_enhancements()
-        redirect = config.redirect if "redirect" in supported else False
-        hsts = config.hsts if "ensure-http-header" in supported else False
-        uir = config.uir if "ensure-http-header"  in supported else False
-        staple = config.staple if "staple-ocsp" in supported else False
+
+        self._verify_all_config_options_supported(config)
```

Comment:
This method raises an error crashing Certbot if the requested enhancement is not supported. I do not think we should do this here. This code will not run until near the very end of Certbot's execution so unfortunately this doesn't make Certbot fail fast.

Instead, I think we should:

1. Change `raise error.Error(msg)` in `_verify_all_config_options_supported` to `logger.warning(msg)`.
2. Call `_very_all_config_options_supported` in this class's `__init__` method. This will make Certbot fail fast as this happens before Certbot has begun any communication with Let's Encrypt